Scripts/modelsystem.py

```diff
@@ -483,23 +501,30 @@ def _export_model_results(self):
     def _generated_tours_mode(self, mode):
         int_demand = pandas.Series(
             0.0, self.zdata_forecast.zone_numbers, name=mode)
+        int_dist = pandas.Series(
+            0.0, self.zdata_forecast.zone_numbers, name=mode)
         for purpose in self.dm.tour_purposes:
             if mode in purpose.modes and purpose.dest != "source":
                 bounds = (next(iter(purpose.sources)).bounds
                     if isinstance(purpose, SecDestPurpose)
                     else purpose.bounds)
                 int_demand[bounds] += purpose.generated_tours[mode]
-        return int_demand
+                int_dist[bounds] += purpose.generated_distance[mode]
+        return int_demand, int_dist / int_demand
     
     def _attracted_tours_mode(self, mode):
-        int_demand = pandas.Series(0, self.zdata_forecast.zone_numbers, name=mode)
+        int_demand = pandas.Series(
+            0, self.zdata_forecast.zone_numbers, name=mode)
+        int_dist = pandas.Series(
+            0.0, self.zdata_forecast.zone_numbers, name=mode)
```

Comment:
Considering the way we use this method (and _attracted_tours_mode()`, I think we should change it into `_generate_tours()`, which would return to dicts with mode as key. In this way, we could reuse the dicts in `_export_model_results` instead of constructing them multiple times.